src/main/scala/firrtl/annotations/Target.scala

```diff
@@ -103,8 +109,16 @@ sealed trait Target extends Named {
 
 object Target {
   def asTarget(m: ModuleTarget)(e: Expression): ReferenceTarget = e match {
-    case w: WRef => m.ref(w.name)
-    case r: ir.Reference => m.ref(r.name)
+    case w: WRef =>
+      val tokens = toTargetTokens(w.name)
```

Comment:
I need to check, but I think you need this if you want to round trip from `ReferenceTarget` to `WRef` to `ReferenceTarget`.

Specifically, it looks like the PR is starting from a `ReferenceTarget`, going to a `WRef` to pass into `create_exps`, and then going back to a `ReferenceTarget` in https://github.com/freechipsproject/firrtl/pull/1220/files#diff-843bb3fb58154aa537a12d5549272685R293. 

It's been a while since I've looked at this...